app/src/main/java/com/globant/practice/domain/service/ProvideGithubApi.java

```diff
@@ -0,0 +1,68 @@
+package com.globant.practice.domain.service;
+
+import com.globant.practice.BuildConfig;
+import com.globant.practice.domain.model.Profile;
+import com.globant.practice.domain.model.Repository;
+import com.globant.practice.domain.model.User;
+import java.util.List;
+import javax.inject.Inject;
+import io.reactivex.Observable;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * Provides the initialisation of retrofit and manages all the possible Api calls
+ * that the application needs.
+ * Created by jonathan.vargas on 5/04/2017.
+ */
+
+public class ProvideGithubApi {
```

Comment:
This class is not necessary indeed. This logic can be implemented in the intector